lib/exfmt/ast/to_algebra.ex

```diff
@@ -584,22 +585,24 @@ defmodule Exfmt.Ast.ToAlgebra do
       # Top level call
       %{stack: [:call]} ->
         args_with_block? = Enum.any?(args, &Util.call_with_block?/1)
-        arg_list = args_to_algebra(args, ctx, parens: args_with_block?, space: !args_with_block?)
+        arg_list = args_to_algebra(args, ctx, parens: args_with_block?, space: !args_with_block?, nest: true)
         concat(name, nest(arg_list, :current))
 
       # Call inside a do end block
       %{stack: [:call, :do | _]} ->
         args_with_block? = Enum.any?(args, &Util.call_with_block?/1)
-        arg_list = args_to_algebra(args, ctx, parens: args_with_block?, space: !args_with_block?)
-        concat(name, nest(arg_list, :current))
+        arg_list = args_to_algebra(args, ctx, parens: args_with_block?, space: !args_with_block?, nest: false)
```

Comment:
This `args_to_algebra/3` function is getting really quite hairy. Could we start splitting it into a few other functions perhaps?

Reply:
I agree that would be a good move. Maybe we could open a separate issue and close/merge this PR for now.

Reply:
@lpil What do you think about this?